src/types/CType.ts

```diff
@@ -24,7 +24,7 @@ export interface ICtypeMetadata {
 }
 
 export default interface ICType {
-  hash: string
+  hash?: string
```

Comment:
We were always back and forth about this being optional. Maybe it is the right choice to have it like this. Why did you have to make it optional?

Reply:
When building from existing CType its hash is already generated and we want to reuse it, if not, it should still be a valid ICType Interface. this might be changed once constructor has all logic outsourced